docs/references/expo/custom-auth-flows/expo-impersonation.mdx

```diff
@@ -0,0 +1,491 @@
+  }, []);
+
+  return actor;
+}
+```
+
+Now our endpoint will call Clerk’s [create actor token](https://clerk.com/docs/reference/backend-api/tag/Actor-Tokens#operation/CreateActorToken) endpoint at `/actor_tokens` and pass in the Clerk secret key for authorization.
+
+```tsx
+// generateActorToken+api.tsx
+
+export async function POST(request: Request) {
+  const body: { user_id: string; actor: { sub: string } } =
+    await request.json();
+
+  const { user_id, actor } = body;
+
+  const res = await fetch("https://api.clerk.com/v1/actor_tokens", {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${process.env.CLERK_SECRET_KEY}`,
+      "Content-type": "application/json",
+    },
+    body: JSON.stringify({
+      user_id,
+      actor: {
+        sub: actor.sub,
+      },
+    }),
+  });
+
+  const data = await res.json();
+
+  return Response.json(data);
+}
+```
+
+## Get the Impersonated user’s data
+
+The second hook will get back user data on the impersonator. You can use this to display UI only the impersonator will see for specific actions, like showing who is currently impersonating and matching the session to the right user for logout. 
+
+```tsx
+// useImpersonatedUser
+
+function useImpersonatedUser(
+  actorSub: string,
+  setImpersonator: React.Dispatch<React.SetStateAction<string | UserDataJSON>>
+) {
+  React.useEffect(() => {
+    const getImpersonatedUser = async () => {
+      const res = await fetch(`/getImpersonatedUser`, {
+        method: "POST",
+        body: JSON.stringify({
+          impersonator_id: actorSub,
+        }),
+      });
+
+      const data = await res.json();
+
+      setImpersonator(data);
+
+      getImpersonatedUser();
+    };
+  }, [actorSub]);
+}
+```
+
+With our hook setup we now can create our API route that will call Clerk’s retrieve user endpoint, `/users`, and get back the Impersonated user’s full `User` object.
+
+```tsx
+// getImpersonatedUser+api.tsx
+
+export async function POST(request: Request) {
+  const body: { impersonator_id: string } = await request.json();
+
+  const { impersonator_id } = body;
+  const res = await fetch(`https://api.clerk.com/v1/users/${impersonator_id}`, {
+    headers: {
+      Authorization: `Bearer ${process.env.CLERK_SECRET_KEY}`,
+    },
+  });
+  const data = await res.json();
+
+  return Response.json(data);
+}
+```
+
+Now that we have the functionality for generating an impersonation session & getting their user data, we have a few more functions to complete that will get the `ticket` from the URL generated by our `useImpersonation` hook, pass that to our `signIn` function to create a new impersonated session, and allow for logout of the impersonated or the impersonator. 
+
+## Generate an impersonation session
+
+Let’s create our functions that work together to generate a new impersonate session, `extractTicket` and `impersonateUser`
+
+```tsx
+  // use this function to get the ticket ID from the response of our generateActorToken API
```

Comment:
````suggestion

### Create functions to generate an impersonated session

Now that you have the functionality for generating an impersonated session & getting the impersonated user's data, there are a few more functions to complete that will:

1. Get the `ticket` from the URL generated by your `useImpersonation` hook
1. Pass the `ticket` to our `signIn` function to create a new impersonated session, allowing you to sign out the impersonated user or impersonator.

The following code should be placed in the `Page` component of your `app/(dashboard)/index.tsx` file:

```tsx filename="app/(dashboard)/index.tsx"
  // use this function to get the ticket ID from the response of our generateActorToken API
````

- Cleaned things up a bit
- Broke a big sentence into a list so it's easier to read